modules/audigentRtdProvider.js

```diff
@@ -26,116 +24,112 @@ const storage = getStorageManager();
 
 /** @type {string} */
 const MODULE_NAME = 'realTimeData';
-
-/** @type {ModuleParams} */
-let _moduleParams = {};
+const SUBMODULE_NAME = 'audigent';
 
 /**
  * XMLHttpRequest to get data form audigent server
  * @param {string} url server url with query params
  */
-
 export function setData(data) {
   storage.setDataInLocalStorage('__adgntseg', JSON.stringify(data));
 }
 
-function getSegments(adUnits, onDone) {
-  try {
-    let jsonData = storage.getDataFromLocalStorage('__adgntseg');
-    if (jsonData) {
-      let data = JSON.parse(jsonData);
-      if (data.audigent_segments) {
-        let dataToReturn = adUnits.reduce((rp, cau) => {
-          const adUnitCode = cau && cau.code;
-          if (!adUnitCode) { return rp }
-          rp[adUnitCode] = data;
-          return rp;
-        }, {});
-
-        onDone(dataToReturn);
-        return;
-      }
+function addSegmentData(adUnits, data) {
+  adUnits.forEach(adUnit => {
+    if (adUnit.hasOwnProperty('bids')) {
+      adUnit.bids.forEach(bid => {
+        bid.audigent_segments = data;
+      })
     }
-    getSegmentsAsync(adUnits, onDone);
-  } catch (e) {
-    getSegmentsAsync(adUnits, onDone);
-  }
+  })
+
+  return adUnits;
 }
 
-function getSegmentsAsync(adUnits, onDone) {
+function getSegments(reqBidsConfigObj, onDone, config, userConsent) {
+  const adUnits = reqBidsConfigObj.adUnits || getGlobal().adUnits;
+
+  let jsonData = storage.getDataFromLocalStorage('__adgntseg');
+  if (jsonData) {
+    let data = JSON.parse(jsonData);
+    if (data.audigent_segments) {
+      addSegmentData(adUnits, data.audigent_segments);
+      onDone();
+      return;
+    }
+  }
+
   const userIds = (getGlobal()).getUserIds();
-  let tdid = null;
+  if (typeof userIds == 'undefined' || userIds == null) {
+    onDone();
+    return;
+  }
 
-  if (userIds && userIds['tdid']) {
-    tdid = userIds['tdid'];
+  if (storage.getDataFromLocalStorage('auHaloId')) {
```

Comment:
you're calling `storage.getDataFromLocalStorage('auHaloId')` twice. I suggest calling it only once, it minifies a lot better. `getDataFromLocalStorage` will not minify and neither will the string `'auHaloId'` so that's a large character count increase. Var names on the other hand minify to 1 character.

```
let haloId = storage.getDataFromLocalStorage('auHaloId');
if (haloId) {
    userIds.haloId = haloId;
    getSegmentsAsync(adUnits, onDone, config, userConsent, userIds);
}
```

Reply:
added var names